rust/lance/src/io/exec/scalar_index.rs

```diff
@@ -452,6 +424,115 @@ impl MaterializeIndexExec {
     }
 }
 
+#[instrument(name = "make_row_ids", skip(mask, dataset, fragments))]
+async fn row_ids_for_mask(
+    mask: RowIdMask,
+    dataset: &Dataset,
+    fragments: &[Fragment],
+) -> Result<Vec<u64>> {
```

Comment:
We are potentially materializing a huge list here but I see now we always were.  In the future this path should definitely be avoided when we have only a block list but I think we already have a TODO for that.

Reply:
That's a good point. I'll definetely be examining the performance here closely soon.